src/pytorch_lightning/utilities/data.py

```diff
@@ -350,17 +382,36 @@ def _dataloader_init_kwargs_resolve_sampler(
     return {"sampler": sampler, "shuffle": False, "batch_sampler": None}
 
 
+def _inject_arg_to_saved(
+    args: List[Any], kwargs: Dict[str, Any], arg_names: List[str], inject_name: str, inject_object: Any
+) -> Tuple[bool, List[Any], Dict[str, Any]]:
+    """Tries to inject a custom argument to a saved list of args and kwargs.
+
+    Returns a tuple indicating success of the operation and modified saved args and kwargs
+    """
+
+    if inject_name in arg_names:
+        inject_index = arg_names.index(inject_name)
+        args = args[:inject_index] + [inject_object] + args[inject_index + 1 :]
+        return True, args, kwargs
+    elif inject_name in kwargs:
+        kwargs[inject_name] = inject_object
+        return True, args, kwargs
+
+    return False, args, kwargs
```

Comment:
What would be an example where we return False?

Shouldn't we include it as a new `kwarg` if the name isn't present already?

If the answer is no: this function should be called `_replace_value_in_saved`

Reply:
You're right the naming wasn't spot on, changed to your suggestion. False will be returned when we cannot find the argument in the argument list. For example, if we want to replace a sampler, but there is no sampler argument in `arg_names`, nor in `kwargs`

Reply:
And yes, the answer is no :smile: We should not include it in `kwargs`, because the method might not support this particular argument